init.lua

```diff
@@ -28,25 +28,33 @@ local function cmpy(p2, y)
 		return math.max(1, p2)
 	end
 end
+
+local function chose_color(pos)
+	-- Calculate color from biome heat and humidity.
+	local biome_data = minetest.get_biome_data(pos)
+	local hum1 = biome_data.humidity
+	local temp1 = biome_data.heat 
+	local vh = math.floor(math.min(math.max(math.floor(hum1), 0), 100) / 6.6)
+	local vt = math.floor(math.min(math.max(math.floor(temp1), 0), 100) / 6.6)
+	vt = vt - math.max(math.min(vt, math.floor(pos.y / 16)), 0)
+	return vh * 16 + vt
+end
+
 local function on_construct(pos)
 	-- get chunk from pos
 	local v = vector.apply(pos, function(a) return math.floor((a - 48) / chunksize) end)
 	local o = vector.subtract(pos, vector.apply(v, function(a) return (a * chunksize) + 48 end))
 	local l = o.z * (chunksize) + o.x
 	local p = minetest.hash_node_position(v)
-
-	local f = io.open(wp .. "/" .. string.format("%d", p), "r")
-	if not f then
-		minetest.log("error", "unable to find map for " .. string.format("%d", p))
-		return
-	end
-
-	local z = f:read("*a")
-	f:close()
-	local map = minetest.decompress(z)
-
+	local map = gt[p]
+	local mv = 0
```

Comment:
The `0` value here is never used. Just use `local mv` here. Luacheck will spot this and warn about it otherwise.